### PR TITLE
deps: upgrade `axe-core` to 4.6.3

### PR DESCRIPTION
**Pull request by adamraine**
https://github.com/dequelabs/axe-core/releases

New rules:
- [`target-size`](https://dequeuniversity.com/rules/axe/4.6/target-size) Seems similar to our tap targets audit 
   - some related discussion in #12345
- [`meta-refresh-no-exceptions`](https://dequeuniversity.com/rules/axe/4.6/meta-refresh-no-exceptions) Looks like a more strict version of [`meta-refresh`](https://dequeuniversity.com/rules/axe/4.6/meta-refresh)
- [`link-in-text-block`](https://dequeuniversity.com/rules/axe/4.6/link-in-text-block) Moved out of experimental. I manually disabled it for now.
- [`frame-title-unique`](https://dequeuniversity.com/rules/axe/4.6/frame-title-unique) [is now wcag2a](https://github.com/dequelabs/axe-core/pull/3610) so we pick it up automatically. I manually disabled for now though.

~There are some smoke failures for `object-alt` and `aria-required-children` that I couldn't figure out just looking at the change-log.~

This should be good now.

cc @jazyan

### CODE REVIEW

**Comment by adamraine** [comment]
I asked about the changes https://github.com/dequelabs/axe-core/issues/3887

File: cli/test/fixtures/a11y/a11y_tester.html [hunk 1]
```diff
@@ -67,8 +67,8 @@
     <section>
       <div
         id="aria-required-children"
-        role="radiogroup">
-        <div></div>
+        role="list">
+        <div>Item</div>
```
Comment (adamraine) [inline cli/test/fixtures/a11y/a11y_tester.html:71]: https://github.com/dequelabs/axe-core/pull/3597

File: cli/test/fixtures/a11y/a11y_tester.html [hunk 2]
```diff
@@ -190,7 +190,7 @@ <h3>sub-sub-header</h3>
     </section>
     <p>object-alt</p>
     <section>
-      <object id="object-alt"></object>
+      <object id="object-alt" data="data:text/html,data"></object>
```
Comment (adamraine) [inline cli/test/fixtures/a11y/a11y_tester.html:193]: https://github.com/dequelabs/axe-core/pull/3680

File: core/gather/gatherers/accessibility.js [hunk 3]
```diff
@@ -61,10 +61,12 @@ async function runA11yChecks() {
       'frame-focusable-content': {enabled: false},
       'aria-roledescription': {enabled: false},
       'scrollable-region-focusable': {enabled: false},
-      // TODO(paulirish): create audits and enable these 3.
+      // TODO(paulirish): create audits and enable these 5.
       'input-button-name': {enabled: false},
       'role-img-alt': {enabled: false},
       'select-name': {enabled: false},
+      'link-in-text-block': {enabled: false},
+      'frame-title-unique': {enabled: false},
```
Comment (connorjclark) [inline core/gather/gatherers/accessibility.js:69]: frame-title-unique seems good to add, if not for 10.0 can you make an issue?

**Review by connorjclark — APPROVED** [review]

**Pull request merged**